### ISSUE TITLE
If Simple App has a book on a spinning hard disk open on shutdown then the book is not re-opened on startup.

### ISSUE THREAD

**Issue by DavidPhillipOster** [MaddTheSane/Simple-Comic#133]
If Simple App has a book  on a spinning hard disk open on shutdown then the book is not re-opened on startup.

## To reproduce: 

* put a .cbz book on a USB3 hard disk that uses spinning platters. 
* Open it in Simple Comic.  With Simple Comic still open, restart the Mac.
* Quickly sign back in.

## Expected behavior:

Simple Comic reopens the .cbz on the correct page.

## Actual behavior:

![file doesn't exist](https://github.com/user-attachments/assets/ed5f037e-7db8-4814-b4ae-980509cd0c64)

Simple Comic shows a "The file doesn't exist" error alert, fails to open the .cbz, and removes the .cbz from its **Open Recent** menu.

## Reproducibility.

I have 3 [WD Elements 6T](https://www.amazon.com/dp/B076MPMZDV) drives on my Mac. I've tried many times and been able to reproduce the bug every single time.

## Analysis:

Simple Comic does a Core Data fetch of the session, which includes bookmark data that should decode to file URLs of the open document. But since Simple Comic is on the internal fast SSD, it is started before the hard disk has a chance to spin up, so resolving the bookmark data to an NSURL fails.  When resolving the bookmark data fails, Simple Comic edits the session managed object, removing all record of that .cbz.

For details on the Bookmark data, see: https://michaellynn.github.io/2015/10/24/apples-bookmarkdata-exposed/ and https://www.mothersruin.com/software/Archaeology/reverse/bookmarks.html

## Pull request

https://github.com/MaddTheSane/Simple-Comic/pull/132

I've made a pull request that on the initial app startup, attempts to resolve the files in the session, without modifying the session. If any fail, it waits 1 second and tries again, up to a total of 5 seconds. In either case it falls through into the exiting code.

## Alternatives

The links above reverse engineered the bookmark data, and with that knowledge it would be possible to extract the file path from the bookmark data to append the NSError with a better message at least showing the path to the file that "didn't exist", but that would not actually solve the problem.

**Comment by DavidPhillipOster** [comment]
I'm currently on macOS Sequoia (v15), but I've had this same problem in Sonoma (v14), Ventura (v13), and Monterey (v12)

**Comment by nickv2002** [comment]
Thanks for the PR and the detailed write up. I'm overdue to push out a bug fix update to the App Store. Will try to find time for that soon.

**Comment by DavidPhillipOster** [comment]
Thank you for evaluating PR so promptly. Thank you for taking the trouble to teach me the standards for this project's PRs. Thank you for letting me play in your sandbox.

**Comment by DavidPhillipOster** [comment]
Further testing, now that this fix is in version 1.9.9.

I had 3 files on the same spinning hard disk open at the time I chose Restart from the Apple menu. After the restart all three were correctly reopened.

With those same three files open, I choose Shutdown from the Apple menu.  On restart, I got one instance of the error alert, but the other two files were opened correctly.

I didn't try repeating these tests to check for reproducible behavior.

So with pull/132, the problem isn't fully solved, only partially solved.
